### PR TITLE
fix: improve http requests RD-1735

### PR DESCRIPTION
**Pull request by doriaviram-lumigo**

### CODE REVIEW

File: src/tracer.js [hunk 1]
```diff
@@ -107,8 +84,6 @@ export const isCallbacked = handlerReturnValue => {
 export const endTrace = async (functionSpan, handlerReturnValue) => {
   try {
     if (functionSpan && !isSwitchedOff() && isAwsEnvironment()) {
-      // eslint-disable-next-line no-undef
-      TracerGlobals.timeoutTimer && clearTimeout(TracerGlobals.timeoutTimer);
```
Comment (efimk-lu) [inline src/tracer.js]: Shouldn't you remove also `clearTimeout`?

File: src/tracer.test.js [hunk 2]
```diff
@@ -39,6 +37,8 @@ describe('tracer', () => {
   spies.clearGlobals = jest.spyOn(globals, 'clearGlobals');
   spies.warnClient = jest.spyOn(logger, 'warnClient');
   spies.logFatal = jest.spyOn(logger, 'fatal');
+  spies.log = jest.spyOn(console, 'log');
```
Comment (efimk-lu) [inline src/tracer.test.js]: Why did you add these spies?

File: src/tracer.js (hunk 1 above)
Comment (doriaviram-lumigo) [reply src/tracer.js]: There is not TracerGlobals.timeoutTimer any more

File: src/tracer.test.js (hunk 2 above)
Comment (doriaviram-lumigo) [reply src/tracer.test.js]: Now he dont print logs when you run npm run test

File: src/tracer.js [hunk 3]
```diff
@@ -61,42 +58,18 @@ export const startTrace = async () => {
...
-    SpansContainer.clearSpans();
-  } else {
-    clearGlobals();
-  }
+  const spansToSend = spans.filter(
+    span => span.transactionId === currentTransactionId
+  );
+  await sendSpans(spansToSend);
+  const hasSpansFromPreviousInvocation = spansToSend.length !== spans.length;
+  hasSpansFromPreviousInvocation && logger.warnClient(LEAK_MESSAGE);
```
Comment (uri-p) [inline src/tracer.js]: Add debug print of the spans from the previous invocation
